### PR TITLE
feat: new site to documentation

### PR DESCRIPTION
**Pull request by emersonlaurentino**
## What's the purpose of this pull request?

Create a new documentation site, which is easier to maintain, but also makes it possible to take advantage of the current documentation that is written in `.mdx`.

Here is an example of the documentation for the Button component.

Screenshot 2022-12-17 at 00 13 02

## How it works?

The new documentation site is made using [Nextra](https://nextra.site/)

## How to test it?

running `yarn workspace site dev`

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+apps/site/*
```
Comment (eduardoformiga) [inline .eslintignore:1]: do we need to ignore the eslint here?

File: apps/site/pages/components/atoms/button.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  <Button variant="primary" size="small">
+    Small
+  </Button>
+</div>
+
+## Styles
+
+```tsx copy
+import '@faststore/ui/src/components/atoms/Button/styles.scss'
+```
```
Comment (eduardoformiga) [inline apps/site/pages/components/atoms/button.mdx:30]: are we gonna add the other sections afterward, right?

File: apps/site/pages/index.mdx [hunk 3]
```diff
@@ -0,0 +1 @@
+Here is where you will have the home page, as there is nothing yet, imagine a beautiful landing page, with copy created by chatGPT.
```
Comment (eduardoformiga) [inline apps/site/pages/index.mdx:1]: hahahahaha 😂

File: apps/site/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (eduardoformiga) [inline apps/site/tsconfig.json:1]: is it something that we can `"extends": "@faststore/shared/tsconfig.json"`? I notice some differences, though.

**Review by eduardoformiga — APPROVED** [review]
Good Job.  🎉 
I left some comments, but it's amazing that we have this initial scaffold.

File: .eslintignore (hunk 1 above)
Comment (emersonlaurentino) [reply .eslintignore:1]: yes, next have the own eslint config, so when we move the core to here, we'll create our config

File: apps/site/pages/components/atoms/button.mdx (hunk 2 above)
Comment (emersonlaurentino) [reply apps/site/pages/components/atoms/button.mdx:30]: yeah, this was for example only

File: apps/site/tsconfig.json (hunk 4 above)
Comment (emersonlaurentino) [reply apps/site/tsconfig.json:1]: then we will create a configuration, similar to eslint, which must be shared between projects

**Review by ArthurTriis1 — APPROVED** [review]

**Pull request merged**